test/test_circular_buffer.jl

```diff
@@ -121,4 +121,17 @@
             @test Array(cb) == [21, 42, 42]
         end
     end
+
+    @testset "_buffer_index" begin
+        cb = CircularBuffer{Int}(5)
+        for i in 1:5
+            push!(cb, i)
+        end
+        k = 1
+        for j in -19:20
+            @test DataStructures._buffer_index(cb, j) == k
+            k += 1
+            if k > 5 k = 1 end
+        end
```

Comment:
This is a little too complex.
Testing the right things,
but a good test set is trivially obvious.

Like 
```
@test cb[1]=1
@test cb[5]=5
@test cb[20]=5
@test cb[0]=5
@test cb[-1]=4
@test cb[-11]=6
```

Your `for` loop does the same thing,
and more comprehensively.
But it is not obvious what it is doing.

Test code that is non-obvious might have bugs.
(and because tests don't have tests ...)